src/Common/src/TypeSystem/Common/TypeSystemConstaintsHelpers.cs

```diff
@@ -47,6 +47,22 @@ private static bool VerifyGenericParamConstraint(Instantiation typeInstantiation
             return true;
         }
 
+        public static bool CheckValidInstantiationArguments(this Instantiation instantiation)
+        {
+            foreach(var arg in instantiation)
+            {
+                if (arg.IsPointer || arg.IsByRef || arg.IsGenericParameter || arg.IsRuntimeDeterminedSubtype || arg.IsVoid)
```

Comment:
This can't stay in this file. It will either block ILVerify and other tools from consuming constraint validation, or maybe it already breaks the build for ILVerify. `IsRuntimeDeterminedSubtype` cannot be used in `/TypeSystem/Common`.

And I would very seriously question why we need this in the first place. If it kicks something out, it means we had invalid instantiations in the system already. I'm not sure what this helps with. We should be weeding them out before any of that reaches these code paths.